packages/editor/src/client/blocks/richTextBlock.tsx

```diff
@@ -431,6 +486,139 @@ function RemoveLinkFormatButton() {
+      </button>
+    )
+  }
+)
+
+function TableButton({icon, iconActive}: TableButtonProps) {
+  const editor = useSlate()
+  const [nrows, setNrows] = useState(2)
+  const [ncols, setNcols] = useState(1)
+
+  const [isTableSpecsOpen, setIsTableSpecsOpen] = useState(false)
+
+  const triggerRef = useRef<any>(null)
+
+  const {t} = useTranslation()
+
+  const open = () => {
+    if (triggerRef.current && !isFormatActive(editor, BlockFormat.Table)) {
+      triggerRef.current!.open()
+      // TODO rather be done with forwardRef ?
+      setIsTableSpecsOpen(true)
+    }
+  }
+  const close = () => {
+    if (triggerRef.current) {
+      triggerRef.current!.close()
+      setIsTableSpecsOpen(false)
+    }
+  }
+
+  const tableSpecs = (
+    <Popover>
+      <div
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          justifyContent: 'space-between',
+          height: '130px'
+        }}>
+        <SetRowColNumbers key={'row'} label="rows" num={nrows} setNumber={setNrows} />
+        <SetRowColNumbers key={'col'} label="columns" num={ncols} setNumber={setNcols} />
+        <Button
+          disabled={isFormatActive(editor, BlockFormat.Table)}
+          onClick={() => {
+            Transforms.insertNodes(editor, emptyCellsTable(nrows, ncols))
+            close()
+          }}>
+          {t('insert table')}
+        </Button>
+      </div>
+    </Popover>
+  )
+
+  const emptyCellsTable = (nrows: number, ncols: number): SlateElement[] => [
+    {
+      type: BlockFormat.Table,
+      children: Array.from({length: nrows}).map(_ => ({
+        type: BlockFormat.TableRow,
+        children: Array.from({length: ncols}).map(_ => ({
+          type: BlockFormat.TableCell,
+          children: [
+            {
+              type: BlockFormat.Paragraph,
+              children: [{text: ''}]
+            }
+          ]
+        }))
+      }))
+    },
+    {
+      type: BlockFormat.Paragraph,
+      children: [{text: ''}]
+    }
+  ]
+
+  return (
+    <>
+      <Whisper placement="top" speaker={tableSpecs} ref={triggerRef} trigger="none">
+        <BaseTableButton
+          icon={isTableSpecsOpen ? iconActive || icon : icon}
+          disabled={isFormatActive(editor, BlockFormat.Table)}
+          active={isTableSpecsOpen}
+          onMouseDown={e => {
+            e.preventDefault()
+            isTableSpecsOpen ? close() : open()
+          }}
+        />
+      </Whisper>
+    </>
+  )
+}
+
+interface SetRowColNumbersProps {
+  label: string
+  num: number
+  setNumber: React.Dispatch<React.SetStateAction<number>>
+}
+
+function SetRowColNumbers({label, num, setNumber}: SetRowColNumbersProps) {
+  const editor = useSlate()
+
+  return (
+    <InputGroup style={{width: '140px'}} disabled={isFormatActive(editor, BlockFormat.Table)}>
+      <InputGroup.Addon style={{width: '60px'}}>{label}</InputGroup.Addon>
+      <Input type="number" value={`${num}`} onChange={val => setNumber(Number(val))} />
```

Comment:
Why do you make the number to a string? and then make it a number again in the `onChange`? I'm pretty sure rsuite has a number input component.

Reply:
I do not find specific number input. Little confusing, the docs state that value prop would be type number:
![image](https://user-images.githubusercontent.com/21113500/100584565-921d5a80-32ec-11eb-8f53-0f77cbb66a18.png)

Reply:
whereas typescript tells, that value is excluseivly string.

Reply:
I do not see a way to do it more elegant.